build.sbt

```diff
@@ -1,31 +1,61 @@
-
-name := "library-example"
+import xerial.sbt.Sonatype._
 
-// also used as a `groupId` by Sonatype
-organization := "ch.epfl.scala"
+lazy val `library-example` = project
+  .in(file("."))
+  .settings(
+    name := "library-example",
+    organization := "ch.epfl.scala",
+    description := "A library that does nothing useful",
+    crossScalaVersions := Seq("3.0.0", "2.13.6", "2.12.14"),
+    scalaVersion := crossScalaVersions.value.head,
+    libraryDependencies += "com.github.scalaprops" %% "scalaprops" % "0.8.3" % Test,
+    testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+  ).settings(publishSettings)
+   .settings(siteSettings)
+   .enablePlugins(MdocPlugin)
 
-libraryDependencies += "com.github.scalaprops" %% "scalaprops" % "0.8.3" % Test
-testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+lazy val publishSettings = Def.settings(
+  sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch")),
+  // indicate the open source licenses that apply to our project
+  licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt")),
+  // publish to the Sonatype repository
+  publishTo := sonatypePublishToBundle.value,
+  // binary compatibility check
+  mimaPreviousArtifacts := Set.empty // Disabled on `master` branch
+)
 
-description := "A library that does nothing useful"
+lazy val root = project
+  .in(file("."))
+  .aggregate(`library-example`)
 
-import xerial.sbt.Sonatype._
-sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch"))
-// indicate the open source licenses that apply to our project
-licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt"))
-// publish to the Sonatype repository
-publishTo := sonatypePublishToBundle.value
+lazy val siteSettings = {
+  /*
+  Combine scaladoc and mdoc.
+  To do so, 
+   1. copy site to target/processed_site
+   2. use mdoc to process site/docs/ **.md
+   3. mdoc outputs new md files in target/processed_site/docs
+   4. scaladoc runs and outputs files in site-ouput
+  */
+  val tmpSiteDir = "processed_site"
+  val siteDir = "site"
+  val outputSiteDir = "site-output"
 
-// documentation website
-enablePlugins(ParadoxPlugin, ParadoxSitePlugin, MdocPlugin, SiteScaladocPlugin)
-mdocIn := sourceDirectory.value / "documentation"
-mdocExtraArguments += "--no-link-hygiene"
-Compile / paradox / sourceDirectory := mdocOut.value
-makeSite := makeSite.dependsOn(mdoc.toTask("")).value
-SiteScaladoc / siteSubdirName := "api"
-paradoxProperties += ("scaladoc.base_url" -> "api")
+  val copySiteTask = Def.task {
+    IO.copyDirectory(
+      baseDirectory.value / "site",
+      target.value / tmpSiteDir
+    )
+  }
 
-// binary compatibility check
-mimaPreviousArtifacts := Set.empty // Disabled on `master` branch
+  Def.settings(
+    Compile / doc / target := baseDirectory.value / outputSiteDir,
```

Comment:
What is the default value for `Compile / doc / target`?

Reply:
It's `target/scala-3.0.0/api`

Reply:
It’s too bad that it contains the full Scala version (although I understand why this is the case) because if we kept the default value, we would have to keep that path up to date in the github workflows when we publish the website (instead of just `site-output`, currently).